### ISSUE TITLE
"panic: runtime error: index out of range" when `podman run` is missing a valid volume mount

### ISSUE THREAD

**Issue by miabbott** [containers/libpod#2118]
/kind bug

**Description**

I fat-fingered the use of `$pwd` (which was unset) when trying to do `podman run` and got a panic.

```shell
$ sudo podman run --name shellcheck --net=host -it -v $pwd:/host registry.access.redhat.com/rhel7
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/containers/libpod/libpod.(*Runtime).newContainer(0xc0000ec240, 0x15359a0, 0xc0000a6048, 0xc000479b80, 0xc000452300, 0x10, 0x10, 0x0, 0x0, 0x0)
        /builddir/build/BUILD/libpod-9551f6bb379d4af56dfb63ddf0f3682e40a6694e/_build/src/github.com/containers/libpod/libpod/runtime_ctr.go:161 +0x1c18
github.com/containers/libpod/libpod.(*Runtime).NewContainer(0xc0000ec240, 0x15359a0, 0xc0000a6048, 0xc000479b80, 0xc000452300, 0x10, 0x10, 0x0, 0x0, 0x0)
        /builddir/build/BUILD/libpod-9551f6bb379d4af56dfb63ddf0f3682e40a6694e/_build/src/github.com/containers/libpod/libpod/runtime_ctr.go:41 +0xe4
main.createContainer(0xc000134840, 0xc0000ec240, 0x0, 0x0, 0x0, 0x0)
        /builddir/build/BUILD/libpod-9551f6bb379d4af56dfb63ddf0f3682e40a6694e/_build/src/github.com/containers/libpod/cmd/podman/create.go:167 +0x2cc
main.runCmd(0xc000134840, 0x0, 0x0)
        /builddir/build/BUILD/libpod-9551f6bb379d4af56dfb63ddf0f3682e40a6694e/_build/src/github.com/containers/libpod/cmd/podman/run.go:53 +0xed
github.com/containers/libpod/vendor/github.com/urfave/cli.HandleAction(0x11a07c0, 0x13fb6d8, 0xc000134840, 0x0, 0xc0002a4600)
        /builddir/build/BUILD/libpod-9551f6bb379d4af56dfb63ddf0f3682e40a6694e/_build/src/github.com/containers/libpod/vendor/github.com/urfave/cli/app.go:501 +0xc8
github.com/containers/libpod/vendor/github.com/urfave/cli.Command.Run(0x135c58b, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1386b1d, 0x20, 0x0, ...)
        /builddir/build/BUILD/libpod-9551f6bb379d4af56dfb63ddf0f3682e40a6694e/_build/src/github.com/containers/libpod/vendor/github.com/urfave/cli/command.go:165 +0x459
github.com/containers/libpod/vendor/github.com/urfave/cli.(*App).Run(0xc000182a80, 0xc0000ae000, 0x9, 0x9, 0x0, 0x0)
        /builddir/build/BUILD/libpod-9551f6bb379d4af56dfb63ddf0f3682e40a6694e/_build/src/github.com/containers/libpod/vendor/github.com/urfave/cli/app.go:259 +0x6bb
main.main()
        /builddir/build/BUILD/libpod-9551f6bb379d4af56dfb63ddf0f3682e40a6694e/_build/src/github.com/containers/libpod/cmd/podman/main.go:263 +0x1577
```

**Describe the results you expected:**
It would be nice if `podman` could detect a bad volume mount flag argument and not panic.

**Additional information you deem important (e.g. issue happens only occasionally):**

A nasty side effect is that `podman run` thinks the container has been created, but it isn't listed in the output of `podman ps`

```shell
$ sudo podman run --name shellcheck --net=host -it -v $(pwd):/host registry.access.redhat.com/rhel7
error creating container storage: the container name "shellcheck" is already in use by "8ed6f8da81a14f2d4917bce763607b47f66929a0134a723208d72645dc5b8911". You have to remove that container to be able to reuse that name.: that name is already in use
$ sudo podman rm shellcheck
unable to find container shellcheck: no container with name or ID shellcheck found: no such container
$ sudo podman rm 8ed6f8da81a14f2d4917bce763607b47f66929a0134a723208d72645dc5b8911
unable to find container 8ed6f8da81a14f2d4917bce763607b47f66929a0134a723208d72645dc5b8911: no container with name or ID 8ed6f8da81a14f2d4917bce763607b47f66929a0134a723208d72645dc5b8911 found: no such container
$ sudo podman ps -a | grep -e shellcheck -e 8ed6f8da81a14f2d49
$ 
```

**Output of `podman version`:**

```
$ podman version
Version:       0.12.1.2
Go Version:    go1.11.2
Git Commit:    "67ab7549b44484cc3f201d7bb2b58b922f8edc24"
Built:         Thu Dec 13 14:35:26 2018
OS/Arch:       linux/amd64
```

**Output of `podman info`:**

```
$ podman info
host:
  BuildahVersion: 1.6-dev
  Conmon:
    package: podman-0.12.1.2-1.git9551f6b.fc29.x86_64
    path: /usr/libexec/podman/conmon
    version: 'conmon version 1.12.0-dev, commit: 67ab7549b44484cc3f201d7bb2b58b922f8edc24'
  Distribution:
    distribution: fedora
    version: "29"
  MemFree: 1182973952
  MemTotal: 16690892800
  OCIRuntime:
    package: runc-1.0.0-66.dev.gitbbb17ef.fc29.x86_64
    path: /usr/bin/runc
    version: |-
      runc version 1.0.0-rc6+dev
      commit: ead425507b6ba28278ef71ad06582df97f2d5b5f
      spec: 1.0.1-dev
  SwapFree: 2114531328
  SwapTotal: 2149576704
  arch: amd64
  cpus: 4
  hostname: mastershake
  kernel: 4.19.12-301.fc29.x86_64
  os: linux
  rootless: true
  uptime: 119h 59m 0.68s (Approximately 4.96 days)
insecure registries:
  registries: []
registries:
  registries:
  - docker.io
  - registry.fedoraproject.org
  - quay.io
  - registry.access.redhat.com
  - registry.centos.org
store:
  ContainerStore:
    number: 0
  GraphDriverName: overlay
  GraphOptions:
  - overlay.mount_program=/usr/bin/fuse-overlayfs
  GraphRoot: /var/home/miabbott/.local/share/containers/storage
  GraphStatus:
    Backing Filesystem: xfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
  ImageStore:
    number: 0
  RunRoot: /run/user/1000

```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

On Fedora Silverblue 29; physical host

**Comment by miabbott** [comment]
Built from master, but still hitting it.  But I see @rhatdan has a PR out...will try that.

```$ bin/podman version
Version:       0.12.2-dev
Go Version:    go1.11.2
Git Commit:    "0f6535cf6b4bfac265983c2fdd3482310ab4f39b"
Built:         Wed Jan  9 21:40:15 2019
OS/Arch:       linux/amd64
$ echo $foo

$ sudo bin/podman run --rm -it -v $foo:/host registry.fedoraproject.org/fedora:29
panic: runtime error: index out of range

goroutine 1 [running]:
github.com/containers/libpod/libpod.(*Runtime).newContainer(0xc0000ec000, 0x1588300, 0xc0000a8038, 0xc0000f4000, 0xc0000f4480, 0x10, 0x10, 0x0, 0x0, 0x0)
        /home/miabbott/workspaces/go/src/github.com/containers/libpod/libpod/runtime_ctr.go:169 +0x1e45
github.com/containers/libpod/libpod.(*Runtime).NewContainer(0xc0000ec000, 0x1588300, 0xc0000a8038, 0xc0000f4000, 0xc0000f4480, 0x10, 0x10, 0x0, 0x0, 0x0)
        /home/miabbott/workspaces/go/src/github.com/containers/libpod/libpod/runtime_ctr.go:41 +0xde
main.createContainerFromCreateConfig(0xc0000ec000, 0xc0000da580, 0x1588300, 0xc0000a8038, 0xc00003cc00, 0x24, 0xc0004f2500)
        /home/miabbott/workspaces/go/src/github.com/containers/libpod/cmd/podman/create.go:919 +0x13e
main.createContainer(0xc00014e840, 0xc0000ec000, 0x0, 0x0, 0x0, 0x0)
        /home/miabbott/workspaces/go/src/github.com/containers/libpod/cmd/podman/create.go:149 +0x1d2
main.runCmd(0xc00014e840, 0x0, 0x0)
        /home/miabbott/workspaces/go/src/github.com/containers/libpod/cmd/podman/run.go:53 +0xed
github.com/containers/libpod/vendor/github.com/urfave/cli.HandleAction(0x11e16e0, 0x1446090, 0xc00014e840, 0x0, 0xc00023e900)
        /home/miabbott/workspaces/go/src/github.com/containers/libpod/vendor/github.com/urfave/cli/app.go:501 +0xc8
github.com/containers/libpod/vendor/github.com/urfave/cli.Command.Run(0x13a5740, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, 0x13d0504, 0x20, 0x0, ...)
        /home/miabbott/workspaces/go/src/github.com/containers/libpod/vendor/github.com/urfave/cli/command.go:165 +0x459
github.com/containers/libpod/vendor/github.com/urfave/cli.(*App).Run(0xc000198c40, 0xc0000b2000, 0x7, 0x7, 0x0, 0x0)
        /home/miabbott/workspaces/go/src/github.com/containers/libpod/vendor/github.com/urfave/cli/app.go:259 +0x6bb
main.main()
        /home/miabbott/workspaces/go/src/github.com/containers/libpod/cmd/podman/main.go:269 +0x15a6
```

**Comment by miabbott** [comment]
#2120 appears to prevent the panic

**Comment by mheon** [comment]
Fixed by #2120
